x-pack/plugins/cases/public/components/header_page/editable_title.tsx

```diff
@@ -47,92 +24,75 @@ const EditableTitleComponent: React.FC<EditableTitleProps> = ({ onSubmit, isLoad
   const { releasePhase, permissions } = useCasesContext();
   const [editMode, setEditMode] = useState(false);
   const [errors, setErrors] = useState<string[]>([]);
-  const [newTitle, setNewTitle] = useState<string>(title);
 
-  const onCancel = useCallback(() => {
-    setEditMode(false);
-    setErrors([]);
-    setNewTitle(title);
-  }, [title]);
+  const onClickSubmit = useCallback(
+    (newTitleValue: string): boolean => {
+      if (!newTitleValue.trim().length) {
+        setErrors([i18n.TITLE_REQUIRED]);
+        return false;
+      }
 
-  const onClickEditIcon = useCallback(() => setEditMode(true), []);
-  const onClickSubmit = useCallback((): void => {
-    if (!newTitle.trim().length) {
-      setErrors([i18n.TITLE_REQUIRED]);
-      return;
-    }
+      if (newTitleValue.trim().length > MAX_TITLE_LENGTH) {
+        setErrors([i18n.MAX_LENGTH_ERROR('title', MAX_TITLE_LENGTH)]);
+        return false;
+      }
 
-    if (newTitle.trim().length > MAX_TITLE_LENGTH) {
-      setErrors([i18n.MAX_LENGTH_ERROR('title', MAX_TITLE_LENGTH)]);
-      return;
-    }
+      if (newTitleValue !== title) {
+        onSubmit(newTitleValue);
```

Comment:
How about?

```suggestion
        onSubmit(newTitleValue.trim());
```

It looks a bit weird that when editing the spaces are displayed but when showing the title they aren't.

https://github.com/elastic/kibana/assets/1533137/e3cb4f3d-81ff-4db5-a80b-7110322dc69b


What was it we decided on this @js-jankisalvi @cnasikas ?

Reply:
I think we discussed to handle it separately because current production behaviour is same as here.
However it is a small change so I am okay if Bree wants to make a change in this PR.